### ISSUE TITLE
Better documentation for developing custom collectors

### ISSUE THREAD

**Issue by obvionaoe** [Cubxity/UnifiedMetrics#77]
### Is there an existing feature request for this?

- [X] I have searched the existing issues before opening this feature request.

### Describe the feature you would like to see.

Improve the documentation for the API used to create custom collectors

### How would you like to see this implemented?

In the wiki or maybe as a JavaDoc

**Comment by Cubxity** [comment]
Hey, have you seen https://github.com/Cubxity/UnifiedMetrics/tree/dev/0.3.x/examples/collector-zombies?

**Comment by obvionaoe** [comment]
Yes, I have, I just think some more docs could be added, like what the purpose of each function is as to allow developers to understand how they should implement them.
Something to help understand why that implementation isn't recommended for production and maybe the difference between async and non async collectors.
I believe you have implemented a good API, but without some extra documentation it might be hard for other developers to understand the purpose of it.
